redbot/core/config.py

```diff
@@ -657,15 +661,39 @@ def __init__(
     def defaults(self):
         return pickle.loads(pickle.dumps(self._defaults, -1))
 
+    @overload
     @classmethod
     def get_conf(
         cls,
-        cog_instance,
+        cog_instance: Cog,
         identifier: int,
-        force_registration=False,
-        cog_name=None,
+        force_registration: bool = False,
+        cog_name: Optional[str] = None,
         allow_old: bool = False,
-    ):
+    ) -> "Config":
+        ...
+
+    @overload
+    @classmethod
+    def get_conf(
+        cls,
+        cog_instance: Optional[Cog],
+        identifier: int,
+        force_registration: bool = False,
+        cog_name: str = "",  # The bool of an empty string is False
```

Comment:
This is sadly the only way I've found to not introduce any breaking changes, since those are args and not kwargs, else `Non-default argument follows default argument` is raised.
I guess it is a fine for solution. For now.
Probably it can be considered to make them positional for Red 3.4? As this imply a breaking change along with discord.py 2.0.